pilot/pkg/serviceregistry/serviceregistry_test.go

```diff
@@ -241,10 +242,13 @@ func TestWorkloadInstances(t *testing.T) {
 		makePod(t, kube, pod)
 		createEndpoints(t, kube, service.Name, namespace, []v1.EndpointPort{{Name: "http", Port: 80}}, []string{pod.Status.PodIP})
 		fx.WaitOrFail(t, "eds")
+		// Endpoint update is triggered since its a brand new service
+		if ev := fx.WaitOrFail(t, "xds full"); !ev.Reason.Has(model.EndpointUpdate) {
+			t.Fatalf("xds push reason does not contain %v: %v", model.EndpointUpdate, ev)
+		}
 		// headless service update must trigger nds push, so we trigger a full push.
-		ev := fx.WaitOrFail(t, "xds full")
-		if !ev.Reason.Has(model.HeadlessEndpointUpdate) {
-			t.Fatalf("xds push reason does not contain %v", model.HeadlessEndpointUpdate)
+		if ev := fx.WaitOrFail(t, "xds full"); !ev.Reason.Has(model.HeadlessEndpointUpdate) {
+			t.Fatalf("xds push reason does not contain %v: %v", model.HeadlessEndpointUpdate, ev)
```

Comment:
This is NOT a production impact change. Only test.

Now our test properly mimics how the real istiod does pushes; in the real world we **are**  triggering double pushes.

Note with debounce its mostly moot, though